provider/pkg/provider/deployment_schedules.go

```diff
@@ -322,24 +323,35 @@ func (st *PulumiServiceDeploymentScheduleResource) Update(req *pulumirpc.UpdateR
 }
 
 func (st *PulumiServiceDeploymentScheduleResource) Read(req *pulumirpc.ReadRequest) (*pulumirpc.ReadResponse, error) {
-	output, err := ToPulumiServiceSharedScheduleOutput(req.GetProperties())
-	if err != nil {
-		return nil, err
-	}
-	input, err := ToPulumiServiceDeploymentScheduleInput(req.GetProperties())
+	stack, scheduleID, err := ParseScheduleID(req.Id, "")
 	if err != nil {
 		return nil, err
 	}
 
-	scheduleID, err := st.client.GetSchedule(context.Background(), output.Stack, output.ScheduleID)
+	scheduleResponse, err := st.client.GetSchedule(context.Background(), *stack, *scheduleID)
 	if err != nil {
 		return nil, fmt.Errorf("failed to read DeploymentSchedule (%q): %w", req.Id, err)
 	}
-	if scheduleID == nil {
-		// if the tag doesn't exist, then return empty response
+	if scheduleResponse == nil {
+		// if schedule doesn't exist, then return empty response
```

Comment:
Might be worth clarifying this comment to state that this deletes the resource from state.
```suggestion
		// if schedule doesn't exist, then return empty response to delete it from state
```

Reply:
Good point, will update all 3 places